### PR TITLE
Add notes from workshop 1 + general details

### PR DESCRIPTION
**Pull request by Ryun1**

### CODE REVIEW

File: CPS-0012/README.md [hunk 1]
```diff
@@ -13,24 +14,68 @@ Created: 2023-11-27
 
 ## Abstract
 
-Cardano lacks a standardized query layer. This leads to suboptimal tooling, dApp and wallet architecture.
+Blockchains are notoriously bad databases, as storage is expensive and queries are complex.
```
Comment (klntsky) [inline CPS-0012/README.md]: Unnecessarily non-neutral IMHO.

File: CPS-0012/README.md [hunk 2]
```diff
@@ -13,24 +14,68 @@ Created: 2023-11-27
...
+This can present issues for builders who wish to build with multiple providers.
+
 ### Query Layers and Tooling
 
 Lack of a standardized query layer results in multiple different implementations of roughly the same set of functionality:
 
 - [Blockfrost](https://blockfrost.io/)
 - [Koios](https://www.koios.rest/)
 - [Maestro](https://www.gomaestro.org)
-- [Ogmios](https://ogmios.dev/)
```
Comment (klntsky) [inline CPS-0012/README.md:27]: Why not Ogmios?

File: CPS-0012/README.md [hunk 3]
```diff
@@ -13,24 +14,68 @@ Created: 2023-11-27
...
 
 As a result, there is a need to support multiple incompatible APIs in downstream tools, examples of which are:
 
 - [Mesh.js](https://meshjs.dev/providers)
 - [Lucid](https://lucid.spacebudz.io/)
 - [cardano-transaction-library](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/runtime.md)
+- [cardano-js-sdk Provider](https://github.com/input-output-hk/cardano-js-sdk/tree/master/packages/core/src/Provider)
```
Comment (klntsky) [inline CPS-0012/README.md]:
```suggestion
- [cardano-js-sdk](https://github.com/input-output-hk/cardano-js-sdk/tree/master/packages/core/src/Provider)
```

File: CPS-0012/README.md [hunk 4]
```diff
@@ -13,24 +14,68 @@ Created: 2023-11-27
 
 ## Abstract
 
-Cardano lacks a standardized query layer. This leads to suboptimal tooling, dApp and wallet architecture.
+Blockchains are notoriously bad databases, as storage is expensive and queries are complex.
+Query layers services abstract away the difficulties of indexing blockchains, offering builders standard API interfaces to access data.
```
Comment (klntsky) [inline CPS-0012/README.md]:
```suggestion
Query layers services abstract away the difficulties of indexing blockchains, offering builders API interfaces to access data. 
```

The next sentence says they are not standardized

**Review by klntsky — CHANGES_REQUESTED** [review]
Great job! Especially the expanded use cases

Just a few notes though

File: CPS-0012/README.md (hunk 2 above)
Comment (Ryun1) [reply CPS-0012/README.md:27]: I dont think Ogmios really applies as it doesnt support historical data

File: CPS-0012/README.md (hunk 1 above)
Comment (Ryun1) [reply CPS-0012/README.md]: fair comment! 😆

Comment (Ryun1) [reply CPS-0012/README.md]: fixed via [2520bca](https://github.com/klntsky/CIPs/pull/1/commits/2520bcaae528099248515747a24390601f69e9c1)

File: CPS-0012/README.md [hunk 5]
```diff
@@ -13,24 +14,68 @@ Created: 2023-11-27
 
 ## Abstract
 
-Cardano lacks a standardized query layer. This leads to suboptimal tooling, dApp and wallet architecture.
+Blockchains lack the properties of classical databases, as storage is expensive and queries are complex.
```
Comment (klntsky) [inline CPS-0012/README.md]: I still don't get what this sentence tries to convey. Which properties, for example?
What queries are complex and why?

Comment (klntsky) [reply CPS-0012/README.md]: Indexers provide access via a relational db and/or a better API than the one node has.

File: CPS-0012/README.md (hunk 2 above)
Comment (klntsky) [reply CPS-0012/README.md:27]: It is not a requirement for a query layer.

Comment (Ryun1) [reply CPS-0012/README.md:27]: yeah fair point

File: CPS-0012/README.md (hunk 5 above)
Comment (Ryun1) [reply CPS-0012/README.md]: in this line and the following im just trying to _very briefly_ introduce why we have query layer providers

File: CPS-0012/README.md [hunk 6]
```diff
@@ -13,24 +14,68 @@ Created: 2023-11-27
...
+Cardano has a range of query layer providers, each with their own interfaces and business motivations.
+This can present issues for builders who wish to build with multiple providers.
+
 ### Query Layers and Tooling
 
 Lack of a standardized query layer results in multiple different implementations of roughly the same set of functionality:
 
 - [Blockfrost](https://blockfrost.io/)
 - [Koios](https://www.koios.rest/)
 - [Maestro](https://www.gomaestro.org)
```
Comment (Ryun1) [inline CPS-0012/README.md:41]:
```suggestion
- [Maestro](https://www.gomaestro.org)
- [Ogmios](https://ogmios.dev/)
```

File: CPS-0012/README.md (hunk 2 above)
Comment (Ryun1) [reply CPS-0012/README.md:27]: changed!

File: CPS-0012/README.md (hunk 5 above)
Comment (Ryun1) [reply CPS-0012/README.md]: @klntsky feel free to remove this, if youd prefer :)

Comment (Ryun1) [reply CPS-0012/README.md]: I have removed

**Comment by Ryun1** [comment]
Messed up the git history, so opened a new PR #2

**Pull request closed without merging**